gettsim/pre_processing/generic_functions.py

```diff
@@ -184,9 +184,9 @@ def create_intercepts(
     for i, up_thr in enumerate(upper_thresholds[:-1]):
         intercepts_at_lower_thresholds[i + 1] = fun(
             up_thr,
```

Comment:
Why not use `x=up_thr` for consistency with the rest?

I do not really want to open a can of worms, but... The docstring of this function is incomprehensible and misleading (`piecewise_linear` does not seem to exist?). Holds for most docstrings in the module at a quick scan.

Also, all functions in here relate to some piecewise function. Why have two separate modules? This one seems superfluous.

Reply:
I will do `x=up_thr`.

I cannot answer the remaining questions.

Reply:
Yes, we can move that all to one module. Sorry about the docstrings! I will revisit them. Once we implemented all the changes for the dag and are sure about the final layout I will revisit! I'll create an issue!

Reply:
https://github.com/iza-institute-of-labor-economics/gettsim/issues/183

Reply:
I was too slow in typing, @MaxBlesch :grinning: